### ISSUE TITLE
Grouped, Stacked Scatter points do not respect minDomain

### ISSUE THREAD

**Issue by MrHen** [FormidableLabs/victory#1279]
# Bugs and Questions

### Checklist

- [x] This is not a `victory-native` specific issue. (Issues that only appear in `victory-native` should be opened [here](https://github.com/FormidableLabs/victory-native/issues/new))

- [x] I have read through the [FAQ](https://formidable.com/open-source/victory/docs/faq) and [Guides](https://formidable.com/open-source/victory/guides/) before asking a question

- [x] I am using the latest version of Victory

- [x] I've searched open issues to make sure I'm not opening a duplicate issue

### The Problem

Setting `minDomain` inside of the Stacked Areas demo shows points outside of the domain.

![Screen Shot 2019-03-26 at 9 14 59 AM](https://user-images.githubusercontent.com/74260/55004202-cda84300-4fa7-11e9-9af4-42ce78ac23d2.png)

### Reproduction

https://codesandbox.io/s/1v5mrvov34

This can also be reproduced at https://formidable.com/open-source/victory/gallery/stacked-areas-with-victory-portal/ using the following code:

```
class App extends React.Component {
  render() {
    return (
      <div>
        <VictoryChart minDomain={{ x: new Date(1991, 1, 1) }} scale={{ x: "time" }} width={400} height={400}>
          <VictoryStack colorScale="warm">
            <VictoryGroup
              data={[
                { x: new Date(1986, 1, 1), y: 2 },
                { x: new Date(1996, 1, 1), y: 3 },
                { x: new Date(2006, 1, 1), y: 5 },
                { x: new Date(2016, 1, 1), y: 4 }
              ]}
            >
              <VictoryArea/>
              <VictoryPortal>
                <VictoryScatter
                  style={{ data: { fill: "black" } }}
                />
              </VictoryPortal>
            </VictoryGroup>
            <VictoryGroup
              data={[
                { x: new Date(1986, 1, 1), y: 4 },
                { x: new Date(1996, 1, 1), y: 3 },
                { x: new Date(2006, 1, 1), y: 2 },
                { x: new Date(2016, 1, 1), y: 5 }
              ]}
            >
              <VictoryArea/>
             <VictoryPortal>
                <VictoryScatter
                  style={{ data: { fill: "black" } }}
                />
              </VictoryPortal>
            </VictoryGroup>
            <VictoryGroup
              data={[
                { x: new Date(1986, 1, 1), y: 3 },
                { x: new Date(1996, 1, 1), y: 1 },
                { x: new Date(2006, 1, 1), y: 4 },
                { x: new Date(2016, 1, 1), y: 2 }
              ]}
            >
              <VictoryArea/>
              <VictoryPortal>
                <VictoryScatter
                  style={{ data: { fill: "black" } }}
                />
              </VictoryPortal>
            </VictoryGroup>
          </VictoryStack>
        </VictoryChart>
      </div>
    );
  }
}

ReactDOM.render(<App/>, mountNode);
```

**Comment by MrHen** [comment]
Quick update, looks like the core issue is just `VictoryScatter`. The following example also shows data outside of the domain:

```
        <VictoryChart minDomain={{ x: new Date(1991, 1, 1) }} scale={{ x: "time" }} width={300} height={300}>
          <VictoryScatter
            data={[
              { x: new Date(1986, 1, 1), y: 2 },
              { x: new Date(1996, 1, 1), y: 3 },
              { x: new Date(2006, 1, 1), y: 5 },
              { x: new Date(2016, 1, 1), y: 4 }
            ]}
            style={{ data: { fill: "black" } }}
          />
        </VictoryChart>
```

**Comment by MrHen** [comment]
Alright, after digging into what was happening, the reason is that the default [`groupComponent`]((https://formidable.com/open-source/victory/docs/victory-scatter/#groupcomponent)) for `VictoryScatter` is `g`. Using [`VictoryClipContainer`](https://formidable.com/open-source/victory/docs/victory-clip-container) changed the behavior so that it trims out anything past the axes.

I'm assuming the reason `VictoryClipContainer` was not used as the default is because a point exactly on an axis should show completely. Using `clipPadding` on the `VictoryClipContainer` worked as expected. This is probably good enough for our purposes but it is kind of annoying.

During the process of digging into this I added a bunch of helpful stories in our fork. [You can take a peek here.](https://github.com/FormidableLabs/victory/compare/master...Centriam:bug/1279-Scatter-minDomain?expand=1) If you want I can submit a PR with them but past that I think this issue can be closed.

**Comment by boygirl** [comment]
@MrHen this expected. `minDomain` / `maxDomain` does not filter data. Components like `VictoryScatter` always plot all of their data even if it overflows the domain. The reason data outside of the domain is not shown for `VictoryArea` in this case is because it uses `VictoryClipContainer` as its `groupComponent` by default. In general you can fix this behavior by changing the `groupComponent` on `VictoryScatter` to `groupComponent={<VictoryClipContainer />}`

**Comment by boygirl** [comment]
I just noticed, though, that it looks like the clipping is not working correctly when the element is rendered within `VictoryPortal`. So I do think this issue should stay open.

**Comment by boygirl** [comment]
Oops, sorry I missed one of your comments

**Comment by boygirl** [comment]
I will also investigate actually filtering the data by domain within the `getData` method that all Victory components use, but I worry that this might have some unintended consequences for things like zooming

**Comment by MrHen** [comment]
I found an example where using `VictoryClipContainer` does not work. If there are two data points that are close enough to each other, then you cannot set a `clipPadding` such that you will fully display the final point (without it being clipped) _and_ fully hide the following point (which is past the defined domain). For instance:

![Screen Shot 2019-04-09 at 2 13 54 PM](https://user-images.githubusercontent.com/74260/55828361-dafa1d00-5ad1-11e9-8d84-2a3849f7249f.png)

**Comment by glimmbo** [comment]
I've got another instance of this, with the labels on my line, on a chart that's got a VictoryZoomContainer. When I load the chart, I focus on more recent points, but the older labels overflow past the domain. I tried the `groupComponent={<VictoryClipContainer />}` on the VictoryLine, but no luck.

![Screen Shot 2020-08-18 at 3 18 55 PM](https://user-images.githubusercontent.com/29473352/90571812-67f1c380-e167-11ea-8662-81d6a53edb5f.png)
